openquake/baselib/tests/performance_test.py

```diff
@@ -0,0 +1,36 @@
+import time
+import unittest
+import tempfile
+import shutil
+from openquake.baselib.performance import Monitor
+
+
+class MonitorTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        tmpdir = tempfile.mkdtemp()
+        cls.mon = Monitor('test', tmpdir)
+
+    def test_no_mem(self):
+        mon = self.mon('test_no_mem')
+        for i in range(3):
+            with mon:
+                time.sleep(0.1)
+        self.assertGreater(mon.duration, 0.3)
+        mon.flush()
+
+    def test_mem(self):
+        mon = self.mon('test_mem', measuremem=True)
+        for i in range(3):
+            with mon:
+                time.sleep(0.1)
+        self.assertGreaterEqual(mon.mem, 0)
```

Comment:
Allocated a list of 100,000 integers, but kept assertGreaterEqual for safety reasons (if the operating system does not see the allocations, we don't want the test to break).